### PR TITLE
d/aws_iam_role Added Tags as an output.

### PR DESCRIPTION
**Pull request by robh007**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
data/aws_iam_role Adds tags as an output to the resource.
```

Output from acceptance testing:

```
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSDataSourceIAMRole'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSDataSourceIAMRole -timeout 120m
=== RUN   TestAccAWSDataSourceIAMRole_basic
=== PAUSE TestAccAWSDataSourceIAMRole_basic
=== RUN   TestAccAWSDataSourceIAMRole_tags
=== PAUSE TestAccAWSDataSourceIAMRole_tags
=== CONT  TestAccAWSDataSourceIAMRole_basic
=== CONT  TestAccAWSDataSourceIAMRole_tags
--- PASS: TestAccAWSDataSourceIAMRole_basic (31.14s)
--- PASS: TestAccAWSDataSourceIAMRole_tags (31.63s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       31.684s


...
```

### CODE REVIEW

File: aws/data_source_aws_iam_role.go [hunk 1]
```diff
@@ -97,6 +99,10 @@ func dataSourceAwsIAMRoleRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	d.Set("unique_id", output.Role.RoleId)
 
+	if len(output.Role.Tags) > 0 {
```
Comment (ewbankkit) [inline aws/data_source_aws_iam_role.go]: @robh007 Thanks for this.
`if len(output.Role.Tags) > 0` shouldn't be necessary here as `keyvaluetags.IamKeyValueTags` can handle that case.

File: aws/data_source_aws_iam_role_test.go [hunk 2]
```diff
@@ -34,6 +34,35 @@ func TestAccAWSDataSourceIAMRole_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSDataSourceIAMRole_tags(t *testing.T) {
```
Comment (ewbankkit) [inline aws/data_source_aws_iam_role_test.go]: Could you also test `resource.TestCheckResourceAttr(dataSourceName, "tags.%", "0"),` in the `TestAccAWSDataSourceIAMRole_basic` case?

**Comment by robh007** [comment]
Updated as per comments from @ewbankkit. Test output below.
```
 make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSDataSourceIAMRole'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSDataSourceIAMRole -timeout 120m
=== RUN   TestAccAWSDataSourceIAMRole_basic
=== PAUSE TestAccAWSDataSourceIAMRole_basic
=== RUN   TestAccAWSDataSourceIAMRole_tags
=== PAUSE TestAccAWSDataSourceIAMRole_tags
=== CONT  TestAccAWSDataSourceIAMRole_basic
=== CONT  TestAccAWSDataSourceIAMRole_tags
--- PASS: TestAccAWSDataSourceIAMRole_basic (29.49s)
--- PASS: TestAccAWSDataSourceIAMRole_tags (29.64s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       29.704s
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSDataSourceIAMRole_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSDataSourceIAMRole_ -timeout 120m
=== RUN   TestAccAWSDataSourceIAMRole_basic
=== PAUSE TestAccAWSDataSourceIAMRole_basic
=== RUN   TestAccAWSDataSourceIAMRole_tags
=== PAUSE TestAccAWSDataSourceIAMRole_tags
=== CONT  TestAccAWSDataSourceIAMRole_basic
=== CONT  TestAccAWSDataSourceIAMRole_tags
--- PASS: TestAccAWSDataSourceIAMRole_tags (22.64s)
--- PASS: TestAccAWSDataSourceIAMRole_basic (23.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	23.613s
```

**Pull request merged**